server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -466,13 +466,75 @@ private void handleIncomingRequest(final HttpRequest httpRequest, final HttpChan
         dispatchRequest(restRequest, channel, badRequestCause);
     }
 
-    private RestRequest requestWithoutContentTypeHeader(HttpRequest httpRequest, HttpChannel httpChannel, Exception badRequestCause) {
+    public static RestRequest createRestRequest(
```

Comment:
I think we can remove these new methods entirely based on the accessiblity of the decompressor, what do you think?

Reply:
In the current state of the companion security PR this is needed to convert an HttpRequest into a RestRequest which is passed into checkAndAuthenticateRequest: https://github.com/opensearch-project/security/pull/3418/files#diff-62eb3270a6f8cc2b9e7b4a92393d1ed9c3ed291ba7548cfddfbe1140e715b0d1R92